src/App.tsx

```diff
@@ -1,61 +1,149 @@
+import { useState, FormEvent } from 'react';
 import './App.scss';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { PreparedTodo } from './types/PreparedTodo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const getNewTodoId = (todoList: PreparedTodo[]) => {
+  const usedIds = todoList.map(todo => todo.id).sort((a, b) => a - b);
+  const freeId = usedIds.findIndex((usId, index) => usId !== index + 1);
+  const newId = freeId !== -1 ? freeId + 1 : usedIds.length + 1;
+
+  return newId;
+};
 
 export const App = () => {
+  const initialTodoList: PreparedTodo[] = todosFromServer.map(todo => ({
+    ...todo,
+    user: usersFromServer.find(user => user.id === todo.userId) || null,
+  }));
+
+  const [titleInput, setTitleInput] = useState('');
+  const [userIdSelect, setUserIdSelect] = useState(0);
+  const [preparedTodoList, setPreparedTodoList] = useState(initialTodoList);
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const handleTitleEnter = (value: string) => {
+    setTitleInput(value);
+    if (titleError === true) {
```

Comment:
titleError - it's a boolean type, so, it's extra
```suggestion
    if (titleError) {
```